src/gui/transferlistwidget.cpp

```diff
@@ -428,65 +428,56 @@ void TransferListWidget::previewSelectedTorrents()
 
 void TransferListWidget::setDlLimitSelectedTorrents()
 {
-    QList<BitTorrent::TorrentHandle *> selected_torrents;
-    bool first = true;
-    bool all_same_limit = true;
+    QList<BitTorrent::TorrentHandle *> TorrentsList;
     foreach (BitTorrent::TorrentHandle *const torrent, getSelectedTorrents()) {
-        if (!torrent->isSeed()) {
-            selected_torrents << torrent;
-            // Determine current limit for selected torrents
-            if (first)
-                first = false;
-            else if (all_same_limit && (torrent->downloadLimit() != selected_torrents.first()->downloadLimit()))
-                all_same_limit = false;
-        }
+        if (torrent->isSeed())
+            continue;
+        TorrentsList += torrent;
     }
+    if (TorrentsList.empty()) return;
 
-    if (selected_torrents.empty()) return;
+    int oldLimit = TorrentsList.first()->downloadLimit();
+    foreach (BitTorrent::TorrentHandle *const torrent, TorrentsList) {
```

Comment:
>I didn't read the link. Can you pinpoint what to look in there?

Title is `QList`, search for `Woe unto you, scribes and Pharisees, hypocrites!`. LOL.

>Does that supposedly performance gain justify this? Will it be noticeable for eg 1000 torrents?

Answer for the second question: no, not even for 1000000 items.
Then first question: it depends on how much you want to use very optimize code, for me, I have no problem converting it over.

>But I worry if we change to std::vector that we might lose COW advantages, or a later reader might not understand why used std containers instead of Qt ones and might try to change it back.

That situation might appear in other context, this one is much simpler, so I think it is safe.

Last, I've a idea to make it a bit more concise, let me code it later.